shardingsphere-proxy/shardingsphere-proxy-backend/src/main/java/org/apache/shardingsphere/proxy/backend/text/admin/mysql/executor/MySQLSetCharsetExecutor.java

```diff
@@ -60,7 +61,13 @@ private Charset parseCharset(final String value) {
             case "utf8mb4":
                 return StandardCharsets.UTF_8;
             default:
-                return Charset.forName(value);
+                try {
+                    return Charset.forName(value);
+                    // CHECKSTYLE:OFF
+                } catch (Exception ex) {
```

Comment:
Why does it need the check?

Reply:
> 

Perhaps you could try executing the statement set names "'utf8'";
It may cause Charset.forName("\"'utf8'\"");

Reply:
But you have handled the quote.

Reply:
> Perhaps you could try executing the statement set name '"utf8"';
> It may cause Charset.forName(""utf8"");

I don't think MySQL will accept `set name '"utf8"'`, we just need to handle the `set name "utf8"` in this PR.

Reply:
> > Perhaps you could try executing the statement set name '"utf8"';
> > It may cause Charset.forName(""utf8"");
> 
> I don't think MySQL will accept `set name '"utf8"'`, we just need to handle the `set name "utf8"` in this PR.

I think if we don't handle this exception it could lead to a decrease in user-friendliness like the following.

![image](https://user-images.githubusercontent.com/76552510/177519139-0855500a-5997-4b4e-a802-bdbb9947f900.png)

I can remove this try-catch if this is not a good change.